pkg/apiserver/api_installer.go

```diff
@@ -286,6 +290,9 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	scope := mapping.Scope
 	nameParam := ws.PathParameter("name", "name of the "+kind).DataType("string")
 	pathParam := ws.PathParameter("path", "path to the resource").DataType("string")
+	exportParam := ws.QueryParameter("export", "strip fields for export").DataType("boolean")
+	exactParam := ws.QueryParameter("exact", "if true, maintain cluster-specific fields (e.g. Namespace)").DataType("boolean")
```

Comment:
For the versioned structure to work, you cannot create explicit parameters for export options... instead create a versioned ExportOptions object like:
https://github.com/brendandburns/kubernetes/blob/kubectl5/pkg/apiserver/api_installer.go#L199
And then add its parameter to the get operation through introspection like:
https://github.com/brendandburns/kubernetes/blob/kubectl5/pkg/apiserver/api_installer.go#L465
(Parameter names and descriptions are obtained from the struct tags)

What this will do is that if in the future you have a new version of the API, where ExportOptions has parameters named "doExport" and "retainClusterSpecific", you can handle either version here.

In the handler itself, you will need to decode the parameters using the versioned structure and then use a converter to convert to the unversioned structure like here:
https://github.com/brendandburns/kubernetes/blob/kubectl5/pkg/apiserver/resthandler.go#L159-L188

(btw, I see that we're not doing that for ListOptions, which kind of defeats the purpose of using a versioned struct)